server/entity/splash_potion.go

```diff
@@ -176,6 +177,11 @@ func (s *SplashPotion) New(pos, vel mgl64.Vec3, yaw, pitch float64, t potion.Pot
 	return splash
 }
 
+// Explode ...
+func (s *SplashPotion) Explode(c block.ExplosionConfig, impact float64) {
+	s.SetVelocity(s.Position().Sub(c.Pos).Normalize().Mul(impact))
```

Comment:
This should also add on to the velocity - it shouldn't override existing velocity.